bsync

```diff
@@ -217,35 +233,82 @@ def rsync_check(sshSrc,dirnameSrc, sshDst,dirnameDst):
 	diff = subprocess.check_output(["rsync"]+args+[rsyncsrc, rsyncdst], universal_newlines=True).split("\n")
 
 	diff.remove("")
-	diff.remove("./")
+	if os.name != "nt":
+		diff.remove("./")
 
 	if len(diff) != 0:
 		sys.exit("Error: rsync_check differences:\n"+str(diff))
 
+# correct the snapshot with records from the original loaded snaphot,
+# effectively reseting the unresolved records to their original value
+def fix_unresolved(ssh,dirname, snapname, unresolved):
+	sbproc = None
+	f = None
+	try:
+		# if we need a ssh shell
+		if ssh != None:
+			cmd = ["cat - > %s" % quote(dirname+"/"+snapname)+".tmp"]
+			sbproc = subprocess.Popen(ssh.getcmdlist()+cmd, stdin=subprocess.PIPE)
+		else:
+			f = open(dirname+"/"+snapname+".tmp", "w")
+			sbproc = subprocess.Popen(["cat", "-"], stdin=subprocess.PIPE, stdout=f)
+
+		if sbproc == None:
+			sys.exit("Error updating a snapshot.")
+
+		# filter out unresolved records from the current snapshot
+		fd = get_snap_fd(ssh, dirname, snapname)
+		gen = fileLineIter(fd)
+
+		record = read_file_record(gen)
+		if record==None: sys.exit("Error reading files from filelist") #should be at least one record (dir root)
+		while record != None:
+			inode,path,type,date,size,perms,rec = record
+			if path not in unresolved:
+				sbproc.stdin.write( rec + b"\0" )
+			record = read_file_record(gen)
+
+		# append unresolved records
+		for rec in unresolved.values():
+			if rec != None:
+				sbproc.stdin.write( rec + b"\0" )
+		sbproc.stdin.flush()
+	finally:
+		# cleanup
+		sbproc.stdin.close()
+		sbproc.wait()
+		if f != None:
+			f.close()
+		if sbproc.returncode != 0:
+			sys.exit("Error in process updating a snapshot.")
+
 # take a snapshot of files states from dir, using find. store it in .bsync-snap-XXXX
 # snap format: inode, path, type, date...
-def make_snapshot(ssh,dirname, oldsnapname, newsnapname):
+def make_snapshot(ssh,dirname, oldsnapname, newsnapname, unresolved):
 	global findformat, findcmdlocal, findcmdremote
 
-	cmd = " %s -fprintf %s '%s'" % (quote(dirname), quote(dirname+"/"+newsnapname), findformat)
+	cmd = [dirname, "-fprintf", dirname+newsnapname, findformat]
 	if oldsnapname!=None:
-		cmd+= " && rm -f "+quote(dirname+"/"+oldsnapname)
+		cmd+= ["&&", "rm", "-f", dirname+oldsnapname]
 	# remove inconsistent newsnap if error in find
-	cmd+= " || ( rm -f "+quote(dirname+"/"+newsnapname)+" && false )"
+	cmd+= ["||", "(", "rm", "-f", dirname+newsnapname, "&&", "false", ")"]
```

Comment:
Consider using os.path.join(dirname, newsnapname)

Reply:
I don't think it's a good idea to change that to path.join. Internally, even on Windows, bsync uses slashes everywhere - because it uses posix tools and because all snapshots contain slashes only. So using path.join at this place would be singular.
However, your remark suggests me that path.join should be removed on the remaining three places too, though it probably doesn't break anything, not even on Windows. I haven't tested bsync on Windows in ssh mode, so I cannot tell.